csrc/host_ir/executor.cpp

```diff
@@ -325,24 +325,38 @@ void HostIrEvaluator::handle(Synchronize* synchronize) {
 }
 
 void HostIrEvaluator::handle(LaunchKernel* launch_kernel) {
-  KernelArgumentHolder args;
+  KernelArgumentHolder args, outputs;
+  bool preallocated_outputs = true;
   for (auto& input : launch_kernel->inputs()) {
     args.push(getKnownConcreteValue(input));
   }
+
+  // If all output buffers are known already, pass them to the executor
+  for (auto& output : launch_kernel->outputs()) {
```

Comment:
```suggestion
  for (Val* output : launch_kernel->outputs()) {
```
https://google.github.io/styleguide/cppguide.html#Type_deduction